### ISSUE TITLE
SLF4J configuration warnings when running tests

### ISSUE THREAD

**Issue by mureinik** [checkstyle/checkstyle#17956]
Taken from https://github.com/checkstyle/checkstyle/pull/17925#issuecomment-3406250106:

When running the tests (e.g., see the CI output from https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=32993&view=logs&j=c902ebb4-c9f8-5f09-4e17-ff78fbbc842e&t=9ca98c81-ff64-58f0-9d03-a23ac1c4a111&l=104), a warning about SLF4J's configuration is seen:
```
[INFO] --- surefire:3.5.4:test (default-test) @ checkstyle ---
[INFO] Surefire report directory: /home/vsts/work/1/s/target/surefire-reports
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
```

Note:
In order to easily reproduce this, there's no need to run the entire CI. The following should be sufficient:
```bash
# Any test should do, this is just an example
mvn clean test -Dtest=LineColumnTest
```

**Comment by mureinik** [comment]
I have a PR ready for this, but as per project guidelines, I don't want to push it until the issue is approved by a mainainer.

If  (once?) it's approved, I'd happily push this PR.

**Comment by romani** [comment]
We need separate issues only functional user facing changes in behavior.
All issue participate in automated release notes generation. So we are a bit demanding to let users have good experience in release notes. Additionally approval process prevents wasting time of contributors, as not all possible to fix and improve and .... .

**Comment by mureinik** [comment]
> Additionally approval process prevents wasting time of contributors, as not all possible to fix and improve and .... .

To clarify - I wasn't dogging this process.
I understand why it's needed, and why the CI even checks this. 
I was just stating that I'm waiting for approval before issuing a PR so it doesn't create any redundant noise in the PR queue, but also make it clear that I have a fix ready, so nobody else will waste their time working on this.
